src/storage/types.rs

```diff
@@ -458,3 +458,6 @@ storage_callback! {
 pub trait StorageCallbackType: Sized {
     fn callback(cb: Callback<Self>) -> StorageCallback;
 }
+pub fn is_external_req(s: &str) -> bool {
```

Comment:
It doesn't look very nice to determine whether a string is an external request...
Consider adding comment about how the function is used, and renaming the parameter to `request_source`

Reply:
It's removed and wrapped in the `RequestInfo` struct, PTAL